File: content/snippets/python/language_basics_environment_variables.py
#!/usr/bin/env python3
import os


def main() -> None:
    mode = os.getenv("APP_MODE", "development")
    port = os.getenv("APP_PORT", "8080")

    print("mode:", mode)
    print("port:", port)
    print("has APP_MODE:", "APP_MODE" in os.environ)


if __name__ == "__main__":
    main()

File: content/snippets/typescript/language_basics_environment_variables.ts
#!/usr/bin/env ts-node

import { setImmediate } from "node:timers";

function main(): void {
  const mode = process.env.APP_MODE ?? "development";
  const port = process.env.APP_PORT ?? "8080";

  console.log("mode:", mode);
  console.log("port:", port);
  console.log("has APP_MODE:", process.env.APP_MODE !== undefined);
}

main();
setImmediate(() => {});
